gtsam/linear/GaussianConditional.cpp

```diff
@@ -347,6 +347,10 @@ namespace gtsam {
 
     VectorValues solution = solve(parentsValues);
     Key key = firstFrontalKey();
+
+    // Check if rng is nullptr, then assign default
+    rng = (rng == nullptr) ? &kRandomNumberGenerator : rng;
```

Comment:
Is this a local or a global default rng?

Reply:
It's not a global default. There is one definition in `GaussianConditional.cpp` and one in `DiscreteConditional.h`. I think we should make it a global default so code like `ShonanAveraging` can also leverage it (they all use the same seed of 42).